tests/h/accounts/views_test.py

```diff
@@ -59,6 +60,145 @@ def invalid_form(errors=None):
     return form
 
 
+class TestToXHRResponse(object):
+    """Unit tests for to_xhr_response()."""
+
+    def test_returns_given_result_if_not_xhr(self, pyramid_request):
+        """
+        If ``request`` isn't an XHR request it returns ``non_xhr_result``.
+
+        The calling view callable passes in the result that it would have
+        returned normally if this were not an XHR request as the
+        ``non_xhr_result`` argument. If the given ``request`` is not an XHR
+        request then ``non_xhr_result`` should just be returned unmodified.
+
+        """
+        pyramid_request.is_xhr = False
+        non_xhr_result = mock.Mock()
+
+        result = views.to_xhr_response(pyramid_request,
+                                       non_xhr_result,
+                                       mock.Mock())
+
+        assert result == non_xhr_result
+
+    def test_returns_form_if_xhr(self, pyramid_request):
+        """
+        If ``request`` is an XHR request it should return the rendered ``form``.
+
+        It should return ``form`` rendered to a ``<form>`` element HTML snippet.
+
+        """
+        pyramid_request.is_xhr = True
+        form = mock.Mock()
```

Comment:
Probably good to get into the habit of limiting the interface you're mocking using `spec_set`:

```python
form = mock.Mock(spec_set=['render'])
```